qudtlib-model/src/main/java/io/github/qudtlib/model/Unit.java

```diff
@@ -27,6 +27,42 @@ public static Definition definition(Unit product) {
         return new Definition(product);
     }
 
+    public static Definition definition(String iriBase, FactorUnits factors) {
+        String localName = factors.streamLocalnamePossibilities().findFirst().get();
```

Comment:
Done.

I think its probably better to implement getLocalname to now use the implementation I had here of getting the first localNamePossibility. This would allow for a single implementation of some of the relatively complex name building logic you created. Using the stream().findFirst() approach means that you should have very little overhead compared to the one-off case. However, this is not a big deal for me either way as long as it works.